src/Authentication/Authentication.Core/Utilities/AuthenticationHelpers.cs

```diff
@@ -446,5 +477,63 @@ public static Task DeleteAuthRecordAsync()
                 File.Delete(Constants.AuthRecordPath);
             return Task.CompletedTask;
         }
+
+        public static PopTokenRequestContext CreatePopTokenRequestContext(IAuthContext authContext)
+        {
+            // Creating a httpclient that would handle all pop calls
+            Uri popResourceUri = GraphSession.Instance.GraphRequestProofofPossession.Uri ?? new Uri("https://graph.microsoft.com/beta/organization"); //PPE (https://graph.microsoft-ppe.com) or Canary (https://canary.graph.microsoft.com) or (https://20.190.132.47/beta/me)
+            HttpClient popHttpClient = new(new HttpClientHandler());
+
+            // Find the WWW-Authenticate header in the response.
+            var popMethod = GraphSession.Instance.GraphRequestProofofPossession.HttpMethod ?? HttpMethod.Get;
+            var popResponse = popHttpClient.SendAsync(new HttpRequestMessage(popMethod, popResourceUri)).Result;
+            var popChallenge = popResponse.Headers.WwwAuthenticate.First(wa => wa.Scheme == "PoP");
+            var nonceStart = popChallenge.Parameter.IndexOf("nonce=\"") + "nonce=\"".Length;
```

Comment:
Parsing WWW-Authenticate is a bit tricky. MSAL has a parser that is better suited. I think Azure SDK folks wanted to provide one as well

Reply:
@bgavrilMS Is there a way to use the MSAL parser even when using Azure Identity?

Reply:
Yes, it is a simple helper class. See https://learn.microsoft.com/en-us/dotnet/api/microsoft.identity.client.authenticationheaderparser?view=msal-dotnet-latest

See an example here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/tests/Microsoft.Identity.Test.Unit/WwwAuthenticateParametersTests.cs#L444